sync/class.jetpack-sync-actions.php

```diff
@@ -440,26 +440,34 @@ static function cleanup_on_upgrade( $new_version = null, $old_version = null ) {
 	}
 
 	static function get_sync_status() {
+		require_once JETPACK__PLUGIN_DIR . 'sync/class.jetpack-sync-wp-replicastore.php';
 		self::initialize_sender();
 
 		$sync_module     = Jetpack_Sync_Modules::get_module( 'full-sync' );
 		$queue           = self::$sender->get_sync_queue();
 		$full_queue      = self::$sender->get_full_sync_queue();
 		$cron_timestamps = array_keys( _get_cron_array() );
 		$next_cron       = $cron_timestamps[0] - time();
+		$store           = new Jetpack_Sync_WP_Replicastore();
+		$checksums       = $store->checksum_all();
 
 		$full_sync_status = ( $sync_module ) ? $sync_module->get_status() : array();
+
 		return array_merge(
 			$full_sync_status,
 			array(
-				'cron_size'            => count( $cron_timestamps ),
-				'next_cron'            => $next_cron,
-				'queue_size'           => $queue->size(),
-				'queue_lag'            => $queue->lag(),
-				'queue_next_sync'      => ( self::$sender->get_next_sync_time( 'sync' ) - microtime( true ) ),
-				'full_queue_size'      => $full_queue->size(),
-				'full_queue_lag'       => $full_queue->lag(),
-				'full_queue_next_sync' => ( self::$sender->get_next_sync_time( 'full_sync' ) - microtime( true ) ),
+				'posts_checksum'        => (string) $checksums['posts'],
+				'comments_checksum'     => (string) $checksums['comments'],
+				'post_meta_checksum'    => (string) $checksums['post_meta'],
```

Comment:
Question about the string cast here: should we be more careful about the meta checksums? Seems like they could be `WP_Error`s, and casting that to string is unexpected:

https://github.com/Automattic/jetpack/blob/e9a665bb1789898480d0376acfb30e4d5805d5dc/sync/class.jetpack-sync-wp-replicastore.php#L724

Reply:
Wouldn't that just end up as an empty string then?

Reply:
Nope, `WP_Error` doesn't seem to have a `toString()` implemented, which means it'll throw a fatal error:

> $a = new WP_Error('test_error', 'test message');
>
> var_dump( (string) $a );
>
> Catchable fatal error: Object of class WP_Error could not be converted to string

Reply:
Oh fun.